pkg/process/checks/pod.go

```diff
@@ -0,0 +1,212 @@
+
+	util, err := kubelet.GetKubeUtil()
+	if err != nil {
+		return nil, err
+	}
+
+	podList, err := util.GetRawLocalPodList()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Collected %d pods", len(podList))
+
+	podMsgs := []*model.Pod{}
+
+	for p := 0; p < len(podList); p++ {
+		// extract pod info
+		podModel := extractPodMessage(&podList[p])
+
+		// insert tags
+		tags, err := tagger.Tag(kubelet.PodUIDToTaggerEntityName(string(podList[p].UID)), collectors.HighCardinality)
+		if err != nil {
+			log.Debugf("Could not retrieve tags for pod: %s", err)
+			continue
+		}
+		podModel.Tags = tags
+
+		// scrub & generate YAML
+		for c := 0; c < len(podList[p].Spec.Containers); c++ {
+			scrubContainer(&podList[p].Spec.Containers[c], cfg)
+		}
+		for c := 0; c < len(podList[p].Spec.InitContainers); c++ {
+			scrubContainer(&podList[p].Spec.InitContainers[c], cfg)
+		}
+		yamlPod, _ := yaml.Marshal(podList[p])
+		podModel.Yaml = yamlPod
+
+		podMsgs = append(podMsgs, podModel)
+	}
+
+	groupSize := len(podMsgs) / cfg.MaxPerMessage
+	if len(podMsgs)%cfg.MaxPerMessage != 0 {
+		groupSize++
+	}
+	chunked := chunkPods(podMsgs, groupSize, cfg.MaxPerMessage)
+	messages := make([]model.MessageBody, 0, groupSize)
+	totalContainers := float64(0)
+	for i := 0; i < groupSize; i++ {
+		totalContainers += float64(len(chunked[i]))
+		messages = append(messages, &model.CollectorPod{
+			HostName:  cfg.HostName,
+			Pods:      chunked[i],
+			GroupId:   groupID,
+			GroupSize: int32(groupSize),
+		})
+	}
+
+	return messages, nil
+}
+
+// scrubContainer scrubs sensitive information in the command line & env vars
+func scrubContainer(c *v1.Container, cfg *config.AgentConfig) {
+	// scrub command line
+	scrubbedCmd, _ := cfg.Scrubber.ScrubCommand(c.Command)
+	c.Command = scrubbedCmd
+	// scrub env vars
+	for e := 0; e < len(c.Env); e++ {
+		// use the "key: value" format to work with the regular credential cleaner
+		combination := c.Env[e].Name + ": " + c.Env[e].Value
+		scrubbedVal, err := log.CredentialsCleanerBytes([]byte(combination))
+		if err == nil && combination != string(scrubbedVal) {
+			c.Env[e].Value = redactedValue
+		}
+	}
+}
+
+// chunkPods formats and chunks the ctrList into a slice of chunks using a specific number of chunks.
+func chunkPods(pods []*model.Pod, chunks, perChunk int) [][]*model.Pod {
+	chunked := make([][]*model.Pod, 0, chunks)
+	chunk := make([]*model.Pod, 0, perChunk)
+
+	for _, p := range pods {
+		chunk = append(chunk, p)
+		if len(chunk) == perChunk {
+			chunked = append(chunked, chunk)
+			chunk = make([]*model.Pod, 0, perChunk)
+		}
+	}
+	if len(chunk) > 0 {
+		chunked = append(chunked, chunk)
+	}
+	return chunked
+}
+
+// extractPodMessage extracts pod info into the proto model
+func extractPodMessage(p *v1.Pod) *model.Pod {
+	// pod medatadata
+	podModel := model.Pod{
+		Name:      p.Name,
+		Namespace: p.Namespace,
+	}
+	if len(p.Annotations) > 0 {
+		podModel.Annotations = make([]string, len(p.Annotations))
+		i := 0
```

Comment:
nit: better to use
```
for i:=0; i<len(p.Annotations); i++{
   ...
}
```
here?

Reply:
it's not really what I want to do, I need to loop on the map to fill my slice, iterating on the slice wouldn't me give me the keys for accessing the map, or am I missing something?

Reply:
Ah, sorry I misread. But I guess if you declare `podModel.Annotations = make([]string, 0, len(p.Annotations)))` and use `append`, you don't need `i` at all, right?